### ISSUE TITLE
InvRef passed to player inventory action callbacks point to the wrong inventory

### ISSUE THREAD

**Issue by ClobberXD** [minetest/minetest#9921]
##### Minetest version

```
all of them
```

##### Summary

The InvRef passed to `allow/on_player_inventory_action` callbacks points to the players inventory, and not the inventory modified by the player. Here are my thoughts:

- This is the exact opposite from my understanding of this callback's documentation:

https://github.com/minetest/minetest/blob/15ba75e4cf1d1b8ceaa9d8ce33dcfdd7dbe80741/doc/lua_api.txt#L4496-L4507

  > `player` (type `ObjectRef`) is the player who modified the inventory `inventory` (type `InvRef`)

- This is very unintuitive, because the player's inventory can already be obtained using `player:get_inventory`. But now, another problem is that there's no way to retrieve the modified inventory, apart from hacky implementations that register an inventory callback for the other inventory. But what if a mod wants to track *all* inventory actions? It's pretty frustrating to have to register inventory action callbacks for all created node and detached inventories in the world, just for this purpose.

- Callbacks that provide half of the information is only half as useful. An inventory (move) action (irrespective of put, take, or move), essentially denotes the transfer of an stack from one inventory to another. The action (put, take, or move), just depends on perspective. So all inventory action should have both a src and dest inventories exposed to Lua (they are already readily-available in C++, btw).

With these points in mind, I'd consider this PR half bug report and half feature request.

### To test

To test the bug, add the following code to a mod, and interact with chest inventories:

```lua
minetest.register_on_player_inventory_action(function(player, action, inventory, info)
	minetest.chat_send_all("*********************************")
	minetest.chat_send_all("action = " .. action)
	minetest.chat_send_all("inventory = " .. dump(inventory:get_location()))
	minetest.chat_send_all("info = " .. dump(info))
	minetest.chat_send_all("*********************************")
end)
```

`inventory` always points to the player inventory, as can be seen from the chat output.

**Comment by raymoo** [comment]
I thought this was only for actions within player inventories? Maybe its functionality was expanded at some point and nobody realized more information needed to be provided to the callback.

**Comment by raymoo** [comment]
Yeah, the documentation for `register_on_player_inventory_action` says that it's only called for events in player inventories.

**Comment by Wuzzy2** [comment]
Take a look at #13181.
